snuba/clickhouse/formatter/query.py

```diff
@@ -201,22 +195,29 @@ def _format_arrayjoin(
 
 
 def _format_limit(
-    query: AbstractQuery, formatter: ClickhouseExpressionFormatterBase
+    query: AbstractQuery, formatter: ExpressionVisitor[str]
 ) -> Optional[StringNode]:
     ast_limit = query.get_limit()
     return (
-        StringNode(f"LIMIT {ast_limit} OFFSET {query.get_offset()}")
-        if ast_limit is not None
+        StringNode(f"LIMIT {ast_limit}")
+        if ast_limit is not None and ast_limit != 1000
```

Comment:
1000 is the default limit the parser adds if the user didn't supply one.

Reply:
It would always just add "LIMIT 1000" to the query (unless a different limit was given by the user), I could remove that it just means every snql query we see on bigquery will end with `LIMIT 1000`. It's just slightly cleaner this way and the only downside is we wouldn't see the limit if the user happens to supply a limit of 1000.

Reply:
I hadn't realized this formatter produces the Query sent directly to ClickHouse. I'll revert this.